resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -393,7 +409,30 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
         val offerId = offer.getId.getValue
         val resources = remainingResources(offerId)
 
-        if (canLaunchTask(slaveId, resources)) {
+        var createTask = canLaunchTask(slaveId, resources)
+        if (hostToLocalTaskCount.nonEmpty) {
```

Comment:
Yes, that is the semantics I was going for.

For the scenario you are mentioning, I'm not sure how that would happen. If there are no executors required, then offers will be declined and this logic will not be exercised. When would an executor be launched on any host?